### DISCUSSION TITLE
Select Default Collection Automatically

### DISCUSSION THREAD

**Discussion by szh** [bitwarden/clients#10760] [Password Manager]
### Code Contribution Proposal

I would like to submit a PR to the Bitwarden clients that will change the default behavior when saving a new password to a family/organization, where a Collection is required to be selected. For example:
![image](https://github.com/user-attachments/assets/830f8681-fb6a-4670-9c01-beebad1432d8)

I would like to have the client automatically select the "Default collection" since it is the only collection available. This would decrease friction when adding passwords to organizations.

More details of this feature request can be found in the [discussion forums](https://community.bitwarden.com/t/default-ownership-and-default-collection-folder-when-creating-new-vault-item/13080). It appears this is a popular request and was at one point on the roadmap but never got implemented.

**Comment by micahblut** [comment]
@szh thanks for the proposal. I think this is a great quality of life improvement, and we'd welcome a contribution in this area.

To make it more generic, here's the logic we think should be implemented:

GIVE: for a given org, the user has "can edit" or higher permissions to only one (1) collection
WHEN: the user selects that org from the ownership dropdown
THEN: default the collection to the single collection the user has "can edit" or higher permissions for.

**Reply by szh** [reply]
Amazing, I'll get to work on this. Can you give me any pointers to which files I should be looking at? If not it's fine, I can look through and find it.

**Reply by micahblut** [reply]
Unfortunately, I cannot help in that regard. I am not a developer, just a product manager 😅

**Reply by djsmith85** [reply]
@szh Thanks for offering your help and willingness to contribute 🎉 

In case you hadn't seen it yet, we have a guide over at https://contributing.bitwarden.com/ to get you started with your development environment.

As a pointer for the changes:
Best place to start is the `(Base)AddEditComponent` located at `libs/angular/src/vault/components/add-edit.component.ts`

This is then extended for each client i.e.: 
- `apps/browser/src/vault/popup/components/vault/add-edit.component.ts`
- `apps/desktop/src/vault/app/vault/add-edit.component.ts``
- `apps/web/src/app/vault/individual-vault/add-edit.component.ts`

Be aware that for the extension UI refresh, planned for later this year, is currently under heavy development. There are new components located under `apps/browser/src/vault/popup/components/vault-v2/add-edit/add-edit-v2.component.ts`, but the @bitwarden/team-vault-dev will be able to assist once you have a PR open.

I hope this helps to get you started :) 

Kind regards,
Daniel

**Reply by szh** [reply]
Thank you so much @djsmith85. I've submitted PR #10806, which I've tested locally in my browser. Does this code also make it's way into the mobile apps?

**Reply by djsmith85** [reply]
Awesome, the code-owning team has been notified.

No this code will not be re-used within the native mobile apps. The repositories are https://github.com/bitwarden/android and https://github.com/bitwarden/ios. A separate PRs would need to be created to add you functionality.

I unfortunately can't give you any pointers there as I am not familiar enough with those codebases.
